config/ompi_fortran_get_alignment.m4

```diff
@@ -132,3 +132,155 @@ EOF
     AS_VAR_POPDEF([type_var])dnl
     OPAL_VAR_SCOPE_POP
 ])
+
+# OMPI_FORTRAN_F08_GET_HANDLE_ALIGNMENT(type, variable to set)
+# ------------------------------------------
+AC_DEFUN([OMPI_FORTRAN_F08_GET_HANDLE_ALIGNMENT],[
+    # Use of m4_translit suggested by Eric Blake:
+    # http://lists.gnu.org/archive/html/bug-autoconf/2010-10/msg00016.html
+    AS_VAR_PUSHDEF([type_var],
+       m4_translit([[ompi_cv_fortran_alignment_$1]], [*], [p]))
+
+    AC_CACHE_CHECK([alignment of Fortran $1], type_var,
+        [OMPI_FORTRAN_MAKE_C_FUNCTION([ompi_ac_alignment_fn], [alignment])
+         # Fortran module
+         cat > conftestf.f90 <<EOF
+module mod_alignment
+type, BIND(C) :: test_mpi_handle
+  integer :: MPI_VAL
+end type test_mpi_handle
+type(test_mpi_handle) :: t1
+type(test_mpi_handle) :: t2
+end module
+
+program falignment
+   use mod_alignment
+   external alignment
+   call alignment(t1,t2)
+end program
+EOF
+
+         # C module
+         if test -f conftest.h; then
+             ompi_conftest_h="#include \"conftest.h\""
+         else
+             ompi_conftest_h=""
+         fi
+         cat > conftest.c <<EOF
+#include <stdio.h>
+#include <stdlib.h>
+$ompi_conftest_h
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+void $ompi_ac_alignment_fn(char *a, char *b)
+{
+    int diff = (int) (b - a);
+    if (diff < 0) diff = -diff;
+    FILE *f=fopen("conftestval", "w");
+    if (!f) exit(1);
+    fprintf(f, "%d\n", diff);
+}
+#ifdef __cplusplus
+}
+#endif
+EOF
+
+         OPAL_LOG_COMMAND([$CC $CFLAGS -I. -c conftest.c],
+             [OPAL_LOG_COMMAND([$FC $FCFLAGS $FCFLAGS_f90 conftestf.f90 conftest.o -o conftest $LDFLAGS $LIBS],
+                  [happy="yes"], [happy="no"])], [happy="no"])
+
+         if test "$happy" = "no" ; then
+             OPAL_LOG_MSG([here is the Fortran program:], 1)
+             OPAL_LOG_FILE([conftestf.f90])
+             AC_MSG_WARN([Could not determine alignment of $1])
+             AC_MSG_WARN([See config.log for details])
+             AC_MSG_ERROR([Cannot continue])
+         fi
+
+         AS_IF([test "$cross_compiling" = "yes"],
+             [AC_MSG_ERROR([Can not determine alignment of $1 when cross-compiling])],
+             [OPAL_LOG_COMMAND([./conftest],
+                 [AS_VAR_SET(type_var, [`cat conftestval`])],
+                 [AC_MSG_ERROR([Could not determine alignment of $1])])])
+
+        unset happy ompi_conftest_h
+        rm -rf conftest*])
+
+    AS_VAR_COPY([$2], [type_var])
+    AS_VAR_POPDEF([type_var])dnl
+])dnl
+
+# OMPI_FORTRAN_GET_COMMON_ALIGNMENT(type, variable to set)
+# ------------------------------------------
+AC_DEFUN([OMPI_FORTRAN_GET_COMMON_ALIGNMENT],[
```

Comment:
There is a *lot* of code duplication between OMPI_FORTRAN_F08_GET_HANDLE_ALIGNMENT and this macro (and possible other macros?).  Should these macros share a common back-end?  At first glance, it looks like the only difference between the two is the Fortran code instantiating the variables/types (and the variable result name, of course).